RO-2476: Fjernet spørsmål om samtykke til innsamling av data

gruble opened the pull request:
Siden vi har sluttet å bruke Google Analytics trenger vi ikke lengre be om samtykke til å samle inn brukerdata.
Denne PR fjerner:

- Spørsmål om samtykke ved første-gangs-kjøring av applikasjonen
- Mulighet til å skru av eller på samtykke under "innstillinger"
- Noen referanser til Google Analytics i innstillinger til prosjektet, som vi har glemt å fjerne tidligere

Siden det ikke lengre var noen "vanlige" innstillinger igjen, men kun "avanserte" innstillinger, har jeg også kuttet ut "avanserte" innstillinger slik at alle innstillinger er synlige med en gang man velger innstillinger i menyen.
Dette fikk en del konsekvenser for visning av versjonsnummer, som ikke ble lastet i tide, så dette måtte skrives om.
Gjenstår:

- [x] teste i app. Ser ut som det er noe tull her. Appen får ikke til å hente observasjoner. Må debugge! 
(ser ut som det kanskje hadde med [denne feilen](https://github.com/NVE/regObs4/commit/f3cd919cbb3d92315b37252202a4efbff8d583f0) å gjøre. Funker nå i hvert fall.

gruble committed:
Fjernet spm om samtykke til innsamling av data

Tatt bort skillet mellom vanlige og avanserte innstillinger i menyen. Fjernet også noe GA-greiersom hang igjen.

gruble committed:
Setter kart-komponent aktiv etter startvegviser er fullført

Dette kallet ble glemt når vi fjernet samtykke-dialogen.

jorgkv commented on src/app/core/services/app-version/app-version.service.ts:
```diff
@@ -1,11 +1,17 @@
-  getAppVersion(): AppVersion {
+  get appVersion$(): Observable<AppVersion> {
+    const appVersion = this.getAppVersion();
+    return of(appVersion).pipe(take(1), shareReplay());
```
Når dette er i en getter så har ikke shareReplay noen veldig fornuftig funksjon. 
Denne koden i getteren vil kjøres hver gang noen spør etter appVersion$, så det er egentlig ikke noe som vil "deles" her selv om shareReplay brukes. Appversjon vil leses fra version.json-fila hver gang getteren kalles.

Det går an å bruke en subject i stedet og bare lese version.json-fila under oppstarten, evt bruke en behavioursubject og importere den via
```
import * as version from 'version.json';
```

gruble replied:
Takk. Bra innspill.

gruble replied:
Angular likte ikke ` import * as version from 'version.json'`.
Men dette funka: `import version from 'version.json'`
Mer info: https://stackoverflow.com/questions/71823871/how-to-fix-angular-yelling-should-not-import-the-named-export-from-json

jorgkv commented on src/app/modules/map/components/map/map.component.ts:
```diff
@@ -159,6 +159,8 @@ export class MapComponent implements OnInit, OnDestroy, AfterViewInit {
       },
     });
+
+    this.isActive = new BehaviorSubject(this.autoActivate);
```
Jeg vet ikke helt hva slags konsekvenser det har, men siden autoActivate er in Input, så kan det hende den ikke er gitt til komponenten enda når constructoren kjøres. Det er nok sånn at constructoren kjøres først, så settes inputene, så kjøres ngOnInit osv. Sjekk evt ut dette i angular-dokumentasjon og life cycle methods.

gruble replied:
Godt poeng. Jeg kjører heller autoaktivering i ngOnInit(), men isActive bør instansieres tidligere, ellers kan componentIsActive() kræsje.

jorgkv commented on src/app/modules/shared/services/logging/file-logging.service.ts:
```diff
@@ -129,8 +130,8 @@ export class FileLoggingService {
           deviceInfoFormatted = `manufacturer = ${deviceInfo.manufacturer}, model = ${deviceInfo.model}, os = ${deviceInfo.operatingSystem}, osVersion = ${deviceInfo.osVersion}, webViewVersion = ${deviceInfo.webViewVersion}`;
         }
       })
-      .finally(() => {
-        const versionInfo = this.appVersionService.getAppVersion();
+      .finally(async () => {
+        const versionInfo = await firstValueFrom(this.appVersionService.appVersion$);
         this.log(
           `Version = ${versionInfo.version}, build = ${versionInfo.buildNumber}, ${deviceInfoFormatted}`,
           null,
```
Kanskje ryddigere å ikke kombinere promise og async?

Enten skrive om metoden til å bare bruke async kode, eller skrive noe a la dette hvis ikke det er så viktig at siste delen kjører i en finally-blokk:

```suggestion
          deviceInfoFormatted = `manufacturer = ${deviceInfo.manufacturer}, model = ${deviceInfo.model}, os = ${deviceInfo.operatingSystem}, osVersion = ${deviceInfo.osVersion}, webViewVersion = ${deviceInfo.webViewVersion}`;
        }
      })
      .then(firstValueFrom(this.appVersionService.appVersion$))
      .then((versionInfo) => 
        this.log(
          `Version = ${versionInfo.version}, build = ${versionInfo.buildNumber}, ${deviceInfoFormatted}`,
          null,
```

Men det er pirk, er ikke så farlig.

gruble committed:
Fikset etter kommentarer fra Jørgen

gruble committed:
Fjernet AppVersionService.

Bruker heller versjonsfila direkte der vi trenger versjonsnummer etter tips fra Jørgen.

jorgkv commented on src/app/pages/user-settings/user-settings.page.ts:
```diff
@@ -28,13 +28,13 @@ const TAPS_TO_ENABLE_TEST_MODE = 7;
-  showAdvanced = false;
   isUpdating = false;
-  version: AppVersion;
+  version$: Observable<AppVersion>;
```
version$ kan sikkert slettes nå?

jorgkv reviewed (approved):
Ser bra ut nå! Bare sjekk den lille kommentaren jeg har før du tar det inn i develop + rebase eller lignende etter develop er fiksa.

gruble committed:
Glemte å slette version$ + package-lock

gruble committed:
Lager nå versjonsfil i develop-bygget

 fordi fila trengs for å kjøre testene
